src/Aquarius.ONE.ClientSDK/Utilities/RestHelper.cs

```diff
@@ -66,6 +69,13 @@ public RestHelper(AuthenticationApi authentication, PlatformEnvironment environm
                 else
                     Event(null, new ClientApiLoggerEventArgs { File = file, EventLevel = EnumEventLevel.Warn, HttpStatusCode = response.StatusCode, ElapsedMs = elapsedMs, Module = "RestHelper", Message = $"PostRestJSONAsync Failed: {endpointURL}" });
 
+                if (_throwAPIErrors && !response.IsSuccessStatusCode)

```

Comment:
With all of these repeated action, it might be better to extract it to a method.